plugin/api.thrift

```diff
@@ -130,6 +130,27 @@ struct Argument {
      * Argument type.
      */
     2: required Type type
+    /**
+     * Annotations defined on this type.
+     *
+     * Note that these are the Thrift annotations listed after the type
+     * declaration in the Thrift file.
+     *
+     * Given,
+     *
+     *   struct User {
+     *     1: required i32 id
+     *     2: required string name
+     *   } (key = "id", validate)
+     *
+     * The annotations will be,
+     *
+     *   {
+     *     "key": "id",
+     *     "validate": "",
+     *   }
+     */
+    3: optional map<string, string> annotations
```

Comment:
Do you mean the annotation on the argument or the type? I would expect the argument annotations here.

```
void foo(
  1: string bar (x = "y")
)
```

Reply:
annotations for custom types are already exposed here https://github.com/thriftrw/thriftrw-go/pull/403/files/672994bbbd6cb0b9302bb91760659f38d4fe3d24#diff-6d5ca9a3f7768e98dd3dbcccfef08c37R50

Reply:
I mean to add annotations for this:

https://github.com/thriftrw/thriftrw-go/pull/403/files/672994bbbd6cb0b9302bb91760659f38d4fe3d24#diff-6d5ca9a3f7768e98dd3dbcccfef08c37R184

but it looks like function arguments and exceptions both use `list<Argument>` as its type in the API.

Reply:
exceptions and functions parameters are both 'arguments' at the AST level so annotations work on both.

```
void foo(
  1: string x (a = "b")
) throws (
  1: GreatSadnessError fail (c = "d")
)
```

Either way, the type you're adding is correct but the docs need to be updated.

Reply:
It looks like the comments in `api.thrift` might be inaccurate and unclear; I'll update them.